extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/KeycloakPolicyEnforcerAuthorizer.java

```diff
@@ -0,0 +1,191 @@
+package io.quarkus.keycloak.pep;
+
+import java.security.Permission;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.inject.Singleton;
+
+import org.keycloak.AuthorizationContext;
+import org.keycloak.adapters.KeycloakDeploymentBuilder;
+import org.keycloak.adapters.authorization.KeycloakAdapterPolicyEnforcer;
+import org.keycloak.adapters.authorization.PolicyEnforcer;
+import org.keycloak.representations.adapters.config.AdapterConfig;
+import org.keycloak.representations.adapters.config.PolicyEnforcerConfig;
+
+import io.quarkus.arc.AlternativePriority;
+import io.quarkus.oidc.OidcConfig;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.runtime.QuarkusSecurityIdentity;
+import io.quarkus.vertx.http.runtime.security.HttpAuthorizer;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
+import io.vertx.ext.web.RoutingContext;
+
+@Singleton
+@AlternativePriority(1)
+public class KeycloakPolicyEnforcerAuthorizer extends HttpAuthorizer {
+
+    private KeycloakAdapterPolicyEnforcer delegate;
+
+    @Override
+    public CompletionStage<SecurityIdentity> checkPermission(RoutingContext routingContext) {
+        VertxHttpFacade httpFacade = new VertxHttpFacade(routingContext);
+        AuthorizationContext result = delegate.authorize(httpFacade);
```

Comment:
This is a blocking operation, we need to make sure we are not on an IO thread. This API will need to be changed to add an easy way to run blocking operations.

Reply:
Can we follow the same approach as you did to the `IdentityProviderManager`? So that we pass an object to authorizers similar to `AuthenticationRequestContext` (maybe rename it to become more generic) in order to run blocking tasks?

Reply:
@pedroigor hi, so given we've agreed `UMA` will be managed differently, how will that be done, will it have its own authorization mechanism as well ? Should we give it a try and have the core authorization mechanism be pluggable/customizabl as needed to meet KV ploicy-enforcer requirements ?
I have to admit I don't quite understand, if you could explain what is too rigid/etc in the core one ? Sorry if I'm missing something...

Reply:
The core one is not rigid, on the contrary, it is quite flexible. I just couldn't find a nice way to plug the policy enforcer capabilities. 

However, that is not Quarkus fault. But the nature of this extension that aims a different authorization model.

If you look at the guide, for instance, you'll see that users can enable the policy enforcer without even specify a single path. Where the mapping between paths and policies are done by the extension itself, through some communication with the Keycloak server.

I see this extension as similar to any other that brings functionality from a 3rd party framework/library. We are also constantly involving the capabilities, adding new ones, etc. We may even end up in a situation where a new capability would require changes in Quarkus.

So, what Stuart did is flexible enough from both a user perspective in regards to declarative authorization (support for different/custom policies, reuse policies across different permissions, etc) as well as from a developer perspective, like in this case, where we are enabling centralized authorization for Quarkus apps.